microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodInvoker.java

```diff
@@ -334,7 +336,17 @@ private CompletableFuture<Object> callSupplierNewThread(FtSupplier<Object> suppl
 
         // Call supplier in new thread
         ClassLoader ccl = Thread.currentThread().getContextClassLoader();
-        CompletableFuture<Object> asyncFuture = Async.create().invoke(() -> {
+        AsyncConfig.Builder asyncBuilder = AsyncConfig.builder();
+
+        // Handle a user-provided executor via @WithExecutor
+        if (introspector.hasWithExecutor()) {
+            WithExecutor withExecutor = introspector.withExecutor();
+            ExecutorService executorService = CDI.current().select(ExecutorService.class, withExecutor).get();
```

Comment:
(Per my earlier comment: I see here you are using it as a qualifier, so indeed this confirms that `@Qualifier` is certainly needed.)